### ISSUE TITLE
[Suggestion] AI "core" block

### ISSUE THREAD

**Issue by TIKIRobo** [tsunamayo/Starship-EVO#2773]
I have always wanted some system of adding friendly AI to a ship and so the idea is that you could place a "core" block that maybe comes in both sizes (brick and block) that can enable fun things in your builds. One thing is in the game currently if you are playing  and an enemy attacks you space station or you for what ever reason you have to run to a near by yoke that is linked to a ship core to turn on all of your turrets, but if you had a AI "core" block your turrets can stay on at all times. or another cool thing is in the future we will have to deal with things like cargo and a ship dedicated to cargo may not be able to hold its own in a fight so you could build small fighters or a frigate that are controlled by the AI core that could follow you and protect you. This way in the future  ships could have a bunch of small swarm AI controlled ships these AI could when damaged try to fly back to the nearest friendly  landing pad to be repaired, if the nearest landing pad so happens to be on your ship then it could act as a carrier and when there are no more enemies your drones could land on your ship and then you could warp away or if they have they're own light speed capabilities they could just follow you instead.

sorry if this is a lot i just kinda felt like it would be interesting to have and defiantly useful in survival  when that happens they could act as automated miners collecting resources and bring them back to your base which is protected by automated turrets etc. 
anyways thanks for reading!

**Comment by ZachZent** [comment]
We will be getting NPC crew in time which will do what you are asking for. Advantages being a more filled in universe rather than just you and a bunch of blocks in a ship.

**Comment by tsunamayo** [comment]
Yes I will add crew and a station block so turret are active. I have several systems to add for stations. Thanks

**Comment by tgiddings** [comment]
NPC crew will probably be best in the majority of situations, but an AI block could still be nice for drone-like ships. You have the opportunity to make drone ships dissimilar in behavior from NPC-crewed ships, such as by only allowing simpler behaviors for drone ships. This gives players a tradeoff between being having a small and simple build with only simple behavior versus a larger and more complex build with more complex behavior available.
